testutils/wallets.go

```diff
@@ -14,3 +16,12 @@ func GenerateWallets(s *require.Assertions, domain *bls.Domain, walletsAmount in
 	}
 	return wallets
 }
+
+func RandomPublicKey() models.PublicKey {
+	publicKey := models.PublicKey{}
+	err := publicKey.SetBytes(utils.RandomBytes(models.PublicKeyLength))
```

Comment:
This method of generating random pubkeys will mostly produce invalid pubkeys, i.e. things that are not points on the G2 curve. Doesn't matter until we do the pairing check (and there we won't have valid signatures anyway), but it could fail some validation checks in Hubble (which are probably missing).

Reply:
good point, I added some more tests and used real pubkeys for them, going forward you're right it's better not to use `RandomPublicKey`